packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2537,8 +2537,12 @@ export class ObjectsInRedisClient {
     // -------------- OBJECT FUNCTIONS -------------------------------------------
     // No callback provided by user, we return a Promise
     private _subscribe(pattern: string | string[], asUser: boolean): Promise<void>;
-    // Callback provided by user, we call callback|
-    private _subscribe(pattern: string | string[], asUser: boolean, callback?: ioBroker.ErrorCallback): void;
+    // Callback provided by user, we call callback
+    private _subscribe<T extends ioBroker.ErrorCallback>(
+        pattern: string | string[],
+        asUser: boolean,
+        callback?: T
+    ): T extends ioBroker.ErrorCallback ? void : Promise<void>;
```

Comment:
I think the first overload can be removed now (assuming the conditional type is ok)